tasks/managed/filter-published-fbc-images/filter-published-fbc-images.yaml

```
@@ -0,0 +1,347 @@
+            exit 1
+        fi
+
+        echo "==============================================================="
+        echo "Filtering Already-Released FBC Fragments via Pyxis Index Query"
+        echo "==============================================================="
+        echo ""
+
+        pyxis_server=$(jq -r '.pyxis.server // "production"' "${DATA_FILE}")
+        target_index=$(jq -r '.fbc.targetIndex // ""' "${DATA_FILE}")
+        
+        if [ -z "$target_index" ]; then
+          echo "ERROR: fbc.targetIndex not found in data file"
+          echo "This task requires targetIndex to query for existing index images"
+          exit 1
+        fi
+        
+        case "$pyxis_server" in
+          production)
+            PYXIS_URL="https://pyxis.api.redhat.com/"
+            ;;
+          stage)
+            PYXIS_URL="https://pyxis.preprod.api.redhat.com/"
+            ;;
+          production-internal)
+            PYXIS_URL="https://pyxis.engineering.redhat.com/"
+            ;;
+          stage-internal)
+            PYXIS_URL="https://pyxis.stage.engineering.redhat.com/"
+            ;;
+          *)
+            echo "ERROR: Invalid pyxis server: $pyxis_server"
+            exit 1
+            ;;
+        esac
+
+        echo "Pyxis server: $PYXIS_URL"
+        echo "Target index: $target_index"
+        echo ""
+
+        echo "Querying Pyxis for existing index images..."
+        index_response=$(curl -s --retry 3 \
+          --cert /tmp/crt \
+          --key /tmp/key \
+          -X GET \
+          "${PYXIS_URL}v1/images?filter=docker_image_id==${target_index}" \
+          -H "Content-Type: application/json" || echo "")
+        
+        if [ -z "$index_response" ]; then
+          echo "⚠️  Empty response from Pyxis (treating as no published indices)"
+          echo "  This means:"
+          echo "    - No index images published yet (expected for first release)"
+          echo "    - Or Pyxis credentials/connectivity issues"
```

Hmm, I don't think this is great. We shouldn't just silently ignore issues with accessing pyxis api. Then we will never know if the filtering stops working completely. So if the curl fails, I think we should fail.

But also, I'm worried that the response may be very big. Isn't that a concern? Maybe we could somehow filter it further down if possible? And will it always return everything at once, there is no pagination?

---

> Hmm, I don't think this is great. We shouldn't just silently ignore issues with accessing pyxis api. Then we will never know if the filtering stops working completely. So if the curl fails, I think we should fail.

agree, I've updated the code to:

  - Retry Pyxis queries 3x with exponential backoff (2s, 4s, 6s)
  - Capture HTTP status + curl exit codes
  - Fail loudly if Pyxis is unreachable (prevents silent re-publishing)
  - Only treat empty .data array as legitimate "first release"

Pipeline now fails-safe: if filtering can't verify what's published, it fails.

---

> But also, I'm worried that the response may be very big. Isn't that a concern? Maybe we could somehow filter it further down if possible? And will it always return everything at once, there is no pagination?

Not a bad idea at all! Thank you! It's also not too complicated to implement, at least a simplified version of the solution. 
What I can do now:

  - Add a 90-day date filter to the Pyxis query
  - Set page_size=1000 to reduce API calls
  - Keep full pagination as a future enhancement (only if monitoring shows it's needed)

This should handle the vast majority of cases. If we see issues in production with catalogs that have >1000 index versions in 90 days, we can implement full pagination then (separate Jira).
WDYT?

---

@mmalina implemented as described

---

Thanks, that looks good. I'm just curious, in your testing, how long would this curl request take? Did it look reasonably fast, or did you have to wait a long time?